### PR TITLE
Fix(eos_designs): Incorrect type for ospf.area in network services keys

### LINKED ISSUE #2996: Synchronize schema for ospf area type in between eos_designs and eos_cli_config_gen
### Issue Summary

Schema type for ```ospf area``` should be the same in role ```eos_designs```and ```eos_cli_congi_gen```.
It is currently ```<int>```versus ```<str>```.

eos_designs:
```
<network_services_keys.name>:
  - name: <str>
    vrfs:
      - name: <str>
        l3_interfaces:
          - interfaces:
            <snip>
            ospf:
              area: <int>
```

eos_cli_config_gen:
```
ethernet_interfaces:
  - name: <str>
  <snip>
    ospf_area: <str>
```

### Which component(s) of AVD impacted

eos_cli_config_gen

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

_No response_

### Relevant log output

```shell
[WARNING]: [KHM-LEAF-202]: 'Validation Error: tenants[0].vrfs[0].l3_interfaces[0].ospf.area': '0.0.0.5' is not of type 'int'
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by carlbuchmann**
## Change Summary

 Incorrect type for ospf.area in network services keys, should be of type string instead of an integer and update default value.

## Related Issue(s)

Fixes #2996

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- updated schema to type str
- update schema default from `0` to `"0"` .
- updated structured config output from `0` to `"0"`

## How to test

- Molecule passes and no changes.

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
we should default to `"0.0.0.0"` since that is how EOS translates `"0"`

**Comment by carlbuchmann** [comment]
> we should default to `"0.0.0.0"` since that is how EOS translates `"0"`

Agreed and updated PR.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlan_interfaces.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def _check_virtual_router_mac_address(vlan_interface_config: dict, variables: li
                 vlan_interface_config["ip_helpers"] = ip_helpers
 
         if get(svi, "ospf.enabled") is True and get(vrf, "ospf.enabled") is True:
-            vlan_interface_config["ospf_area"] = svi["ospf"].get("area", 0)
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlan_interfaces.py:154]: There is no schema update for this change.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml [hunk 2]
```diff
@@ -556,11 +556,11 @@ $defs:
                           type: bool
                           default: false
                         area:
-                          type: int
+                          type: str
                           convert_types:
-                            - str
+                            - int
                           description: OSPF area id.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml]:
```suggestion
                          description: OSPF area ID.
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlan_interfaces.py (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlan_interfaces.py:154]: Schema already set as a string, including default value:

```yaml
$defs:
  svi_settings:
    keys:
      ospf:
        keys:
          area:
            type: str
            convert_types:
              - int
            description: OSPF area ID.
            default: "0"
```

**Review by ClausHolbechArista — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]

**Pull request merged**